### DISCUSSION TITLE
Help with horizontal scroll for custom toolbar and slideshow layout

### DISCUSSION THREAD

**Discussion by JGK1213** [igordanchenko/yet-another-react-lightbox#162] [Q&A] [answered]
Hi! Just want to start off by saying thank you for creating this and for all the great resources and documentation!

I've been able to do so much customization, but I'm stuck on a couple things.

1. I added a custom toolbar that allows for different categories, the user can click on the category and the carousel will scroll to the specific index. I have that working. The part I can't figure out is how to add a horizontal scroll in the toolbar for mobile view. I tried following this, https://github.com/igordanchenko/yet-another-react-lightbox/discussions/146#discussioncomment-6338067, but doesn't seem to add back the horizontal scroll when I add this `plugins={[({ remove }) => remove("no-scroll")]}`. 

2. What's the best way to update the width of the carousel to allow for the layout in the screenshots? Where the current active image shows, but also the prev and next show slightly on either side? I was messing around with the width css, but just curious if you have a good approach in mind.

Here are screenshots for clarity:

desktop:
![Screenshot 2023-07-27 at 2 17 04 AM](https://github.com/igordanchenko/yet-another-react-lightbox/assets/8745658/e51cc19b-6619-4a54-88e6-fc9b2d68f77a)

mobile:
![Screenshot 2023-07-27 at 2 16 23 AM](https://github.com/igordanchenko/yet-another-react-lightbox/assets/8745658/fd08bc17-77ac-4ff2-8d11-8c2a6001db70)

Thank you!

**Comment by igordanchenko** [comment, marked as answer]
Hi there!

I'm glad to hear you are enjoying this library so far.

> I added a custom toolbar that allows for different categories, the user can click on the category and the carousel will scroll to the specific index. I have that working. The part I can't figure out is how to add a horizontal scroll in the toolbar for mobile view. I tried following this, https://github.com/igordanchenko/yet-another-react-lightbox/discussions/146#discussioncomment-6338067, but doesn't seem to add back the horizontal scroll when I add this plugins={[({ remove }) => remove("no-scroll")]}.

I'm not sure that discussion is relevant to your case. Let me explain. In the case when your page content is larger than the viewport height, some browsers (e.g. on Windows) display vertical scrollbar. So when the lightbox opens, that vertical scrollbar remains visible but it makes no sense since the lightbox takes over fullscreen user experience. So when the lightbox opens, it hides that scrollbar by setting document `<body>` height to `100%` and overflow to `hidden`.

But in this case you probably want to limit the horizontal dimension of your toolbar and set overflow to `auto`. Please see the below example. It works fine on iOS, however I'm not sure how well it's going to work on desktop browsers on Windows (as they may be showing horizontal scrollbar).

https://codesandbox.io/p/sandbox/yet-another-react-lightbox-162-98pclr

> What's the best way to update the width of the carousel to allow for the layout in the screenshots? Where the current active image shows, but also the prev and next show slightly on either side? I was messing around with the width css, but just curious if you have a good approach in mind.

To achieve this effect, I'd wrap carousel in another `<div>` and give it some extra margin. You'd also need to adjust the position of navigation buttons and the toolbar. I'm assuming your images are of the same aspect ratio, as this layout may look somewhat odd otherwise. Please see the below example.

https://codesandbox.io/p/sandbox/yet-another-react-lightbox-162-98pclr

I hope this helps.

**Reply by JGK1213** [reply]
thanks for the quick reply! I'll take a look at those examples you added and go from there!
